scipy/signal/spectral.py

```diff
@@ -1246,12 +1387,15 @@ def istft(Zxx, fs=1.0, window='hann', nperseg=None, noverlap=None, nfft=None,
         x[..., ii*nstep:ii*nstep+nperseg] += xsubs[..., ii] * win
         norm[..., ii*nstep:ii*nstep+nperseg] += win**2
 
-    # Divide out normalization where non-tiny
-    x /= np.where(norm > 1e-10, norm, 1.0)
-
     # Remove extension points
     if boundary:
         x = x[..., nperseg//2:-(nperseg//2)]
+        norm = norm[..., nperseg//2:-(nperseg//2)]
+
+    # Divide out normalization where non-tiny
+    if np.sum(norm > 1e-10) != len(norm):
+        warnings.warn("NOLA condition failed, STFT may not be invertible")
```

Comment:
This takes care of my comment above about checking NOLA, right?

Reply:
Yes